src/openssl_tls/transport.rs

```diff
@@ -0,0 +1,189 @@
+#![cfg(feature = "tls-openssl")]
```

Comment:
There is `#![cfg(feature = "tls-openssl")]` at the top of `openssl_tls/mod.rs`, is this cfg necessary?